### PR TITLE
Adding support for touch events

### PR DESCRIPTION
**Pull request by malthe**
This adds support for the following events:
- `ontouchstart`
- `ontouchend`
- `ontouchmove`
- `ontouchcancel`

### CODE REVIEW

File: lib/svg_sigs.mli [hunk 1]
```diff
@@ -673,7 +673,7 @@ module type T = sig
   val a_onunload : Xml.event_handler  -> [> | `OnUnload ] attrib
   val a_onzoom : Xml.event_handler  -> [> | `OnZoom ] attrib
 
-  (** {3 Javascript mouse events} *)
+  (** {6 Javascript mouse events} *)
```
Comment (Drup) [inline lib/svg_sigs.mli]: Why changing the header level ?

**Comment by Drup** [comment]
@vasilisp I suppose you guys already have a custom implem of that. Anything to add ? :)

**Comment by vasilisp** [comment]
@Drup we clearly do use the touch events, but via `Lwt_js_events`, so I had not even noticed that they are missing in TyXML. The PR looks good (modulo your comments) :).

**Comment by Drup** [comment]
I'm ... not quite sure what's happening in CI.

**Comment by vasilisp** [comment]
A missing `type Xml.touch_event_handler = Xml.touch_event_handler` constraint in the `T with ...`?

**Comment by Drup** [comment]
> A missing type Xml.touch_event_handler = Xml.touch_event_handler constraint in the T with ...?

Ah! Well caught.

@malthe You need to go through the various `with` constraints and every time you see events, add the touch one.

File: lib/svg_sigs.mli (hunk 1 above)
Comment (malthe) [reply lib/svg_sigs.mli]: Oh lol – I obviously didn't get what that was at all.

**Comment by malthe** [comment]
@Drup I don't know if I know how to write/add PPX tests. I'm an OCaml newbie, relatively speaking ...

File: lib/html_sigs.mli [hunk 2]
```diff
@@ -1141,6 +1141,7 @@ sig
      and type Xml.event_handler = Xml.event_handler
      and type Xml.mouse_event_handler = Xml.mouse_event_handler
      and type Xml.keyboard_event_handler = Xml.keyboard_event_handler
+     and type Xml.touch_event_handler = Xml.touch_event_handler
```
Comment (Drup) [inline lib/html_sigs.mli]: You should need the same thing on the svg side.

Comment (malthe) [reply lib/html_sigs.mli]: I have them there – just missed one location.

**Comment by Drup** [comment]
For tests, I would like at least one test exercising events with the ppx here: https://github.com/ocsigen/tyxml/blob/master/test/test_ppx.ml#L150 . 
Otherwise, feel free to add new tests in non-ppx syntax here: https://github.com/ocsigen/tyxml/blob/master/test/test_html.ml
You can follow the patterns from the other tests.

**Comment by sagotch** [comment]
> I'm ... not quite sure what's happening in CI.

It looks like a bug in `dune 1.1.0`. I see that `1.1.1` is installed, but console shows `jbuilder` invocation. Could it be using outdated code?

File: ppx/reflect/reflect.ml [hunk 3]
```diff
@@ -163,6 +163,7 @@ let rec to_attribute_parser lang name = function
   | [[%type: Xml.event_handler]]
   | [[%type: Xml.mouse_event_handler]]
   | [[%type: Xml.keyboard_event_handler]] ->
+  | [[%type: Xml.touch_event_handler]] ->
```
Comment (Drup) [inline ppx/reflect/reflect.ml]: There is a syntax error here. You still have the arrow on the line above.

**Comment by Drup** [comment]
@malthe CI should now operational again. I'll merge this as soon as you have fixed the issue in reflect.ml.

**Comment by Drup** [comment]
Thanks!

**Pull request merged**